ODIN_II/SRC/netlist_analyzer.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "netlist_analyzer.h"
+
+int calculate_multiplier_aware_critical_path(nnode_t* node, netlist_t* netlist) {
```

Comment:
merge this bit with the statistic analyzer that's there now

Reply:
Thank you for this suggestion!

Reply:
while you are there, if theres any bit you feel could be improved, feel free to submit that as seperate PR's such that they are easier to merge in and there are fewer moving pieces merge in at once

Reply:
> Thank you for this suggestion!

np :smile: Eager to see this in as it is a super useful addition alongside @sdamghan PR, it would seriously improve the flexibility of the tool!